dbt/adapters/athena/impl.py

```diff
@@ -56,59 +56,72 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     # TODO: Add more lf-tag unit tests when moto supports lakeformation
     # moto issue: https://github.com/getmoto/moto/issues/5964
     @available
-    def add_lf_tags(self, database: str, table: str = None, lf_tags: Dict[str, str] = None):
+    def add_lf_tags(
+        self,
+        database: str,
+        table: str = None,
+        lf_tags: Optional[Dict[str, str]] = None,
+        lf_tags_columns: Optional[Dict[str, Dict[str, List[str]]]] = None,
+    ):
         conn = self.connections.get_thread_connection()
         client = conn.handle
 
         lf_tags = lf_tags or conn.credentials.lf_tags
-        if not lf_tags:
-            logger.debug("No LF tags configured")
-            return
-
-        resource = {
-            "Database": {"Name": database},
-        }
-
-        if table:
-            resource = {
-                "Table": {
-                    "DatabaseName": database,
-                    "Name": table,
-                }
-            }
 
-        with boto3_client_lock:
-            lf_client = client.session.client(
-                "lakeformation", region_name=client.region_name, config=get_boto3_config()
-            )
+        if not lf_tags and not lf_tags_columns:
+            logger.debug("No LF tags configured")
+        else:
+            with boto3_client_lock:
+                lf_client = client.session.client(
+                    "lakeformation", region_name=client.region_name, config=get_boto3_config()
+                )
 
-        response = lf_client.add_lf_tags_to_resource(
-            Resource=resource,
-            LFTags=[
-                {
-                    "TagKey": key,
-                    "TagValues": [
-                        value,
-                    ],
-                }
-                for key, value in lf_tags.items()
-            ],
-        )
+            if lf_tags:
+                resource = {"Database": {"Name": database}}
+                if table:
+                    resource = {"Table": {"DatabaseName": database, "Name": table}}
 
-        failures = response.get("Failures", [])
-        tbl_appendix = f".{table}" if table else ""
-        if failures:
-            base_msg = f"Failed to add LF tags: {lf_tags} to {database}" + tbl_appendix
-            for failure in failures:
-                tag = failure.get("LFTag", {}).get("TagKey")
-                error = failure.get("Error", {}).get("ErrorMessage")
-                logger.error(f"Failed to set {tag} for {database}" + tbl_appendix + f" - {error}")
-            raise DbtRuntimeError(base_msg)
-        else:
-            logger.debug(f"Added LF tags: {lf_tags} to {database}" + tbl_appendix)
+                response = lf_client.add_lf_tags_to_resource(
+                    Resource=resource, LFTags=[{"TagKey": key, "TagValues": [value]} for key, value in lf_tags.items()]
+                )
+                self.parse_lf_client_response(response, database, table, None, lf_tags)
+
+            if lf_tags_columns:
```

Comment:
shall we add a lf_tags_columns validation here? before we "blindly" try to iterate to the dict of dict with columns?

Reply:
you're right. I've added simple config validation here